plugins/inputs/modbus/modbus.go

```diff
@@ -159,6 +167,14 @@ func (m *Modbus) Init() error {
 		return fmt.Errorf("device name is empty")
 	}
 
+	if m.Retries < 0 {
+		return fmt.Errorf("retries cannot be negative")
+	}
+
+	if m.Retries > 0 && m.RetriesWaitTime.Duration < 0 {
+		return fmt.Errorf("wait time between retries cannot be negative")
+	}
```

Comment:
we might want to check that RetriesWaitTime * Retries is not greater than the time the input has to collect the metrics, 1s, I think.

Reply:
Hmm this might be challenging if the time-frame is really 1s. There might be cases with a slow but busy device where come close to this wait time... However it is worth a trial (at least for all other cases). Is the 1 second a fixed value or can I get it from somewhere in the code?

Reply:
Ok I looked into this a bit more, it's not actually 1s per se; it logs "did not complete within its interval" if the runtime is longer than the agent interval setting. 
We could warn if the `RetriesWaitTime * Retries > interval`, where interval is the input's configured interval.

Reply:
Agh. we might not be exposing that setting to the plugin. May not be possible.. Ok maybe just a note in the settings not to pick a value that would end up retrying longer than your interval.

Reply:
See commit(s) below.